static_report/src/utils/index.tsx

```diff
@@ -134,3 +136,113 @@ export function extractExpectedOrActual(value) {
 
   return value;
 }
+
+// for comparison
+export function joinBykey(base = {}, input = {}) {
```

Comment:
This name is a little confusing. Maybe rename as `nestComparisonValueByKey` ? (not the best, but I at least have more context)
Or at least explain more in the comment, as it's not obvious